skimage/feature/peak.py

```diff
@@ -118,13 +152,21 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
     array([[10, 10, 10]])
 
     """
+    out = np.zeros_like(image, dtype=np.bool)
+
+    # no peak for a trivial image
+    if np.all(image == image.flat[0]):
```

Comment:
Trial images are definitely an "edge case", is this check really necessary? the call to `np.all` and checking equality for large images is actually quite expensive. Since this is an "unlikely" scenerio, is it better to move this logic to the end or remove it all together.

Reply:
Moving the code to the end does not help, as it will still need to be checked before final results are returned (potentially overwrite peak candidates).
I am okay with removing the logic, just we might need to somehow documente that there could be an incompatibility for edge cases.

Here is an example, where there was no peak before, it will become 25 peaks.  I agree this is a case probably users should check, instead of checking it every time within the library.

```
        image = np.ones((5, 5))
        # return 0, will return 25 if we remove the trivial case check
        print(len(peak.peak_local_max(image, min_distance=0, threshold_abs=0, indices=True)))

```

Reply:
OK, thanks for the explanation. I guess this can be left as a secondary issue.